render/fx_renderer/fx_pass.c

```diff
@@ -516,19 +470,16 @@ void fx_render_pass_add_box_shadow(struct fx_gles_render_pass *pass,
 
 	set_proj_matrix(renderer->shaders.box_shadow.proj, pass->projection_matrix, &shadow_box);
 	glUniform4f(renderer->shaders.box_shadow.color, color->r, color->g, color->b, color->a);
-	glUniform1f(renderer->shaders.box_shadow.blur_sigma, blur_sigma);
+	glUniform1f(renderer->shaders.box_shadow.blur_sigma, shadow_data->blur_sigma);
 	glUniform1f(renderer->shaders.box_shadow.corner_radius, options->corner_radius);
 	glUniform2f(renderer->shaders.box_shadow.size, shadow_box.width, shadow_box.height);
 	glUniform2f(renderer->shaders.box_shadow.position, shadow_box.x, shadow_box.y);
 
-	render(&shadow_box, &render_region, renderer->shaders.box_shadow.pos_attrib);
-
-	pixman_region32_fini(&render_region);
+	// TODO: properly remove area we don't need to be damaged
+	render(&shadow_box, options->clip, renderer->shaders.box_shadow.pos_attrib);
```

Comment:
We need this before merging. I'll have a crack at this while your timezone is sleeping if you don't mind :D